### PR TITLE
privateer can travel to room without vision

### PR DESCRIPTION
**Pull request by karlthepagan**
this is extremely useful when settling a new area (portals etc)

### CODE REVIEW

**Comment by karlthepagan** [comment]
I just realized this was for privateer not pioneer.

Doesn't change the feature ;) this is what I intended.

File: creep.behaviour.privateer.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ mod.exploitNextRoom = function(creep){
     // no new flag
     // go home
     Population.registerCreepFlag(creep, null);
-    Creep.action.travelling.assign(creep, Game.rooms[creep.data.homeRoom].controller);
+    creep.data.travelRoom = creep.data.homeRoom;
```
Comment (chobobobo) [inline creep.behaviour.privateer.js]: By manually assigning this, are we externally hacking at the internals of the travelling action?  Ultimately, shouldn't creep.action.travelling.assign be able to take travelRoom as a parameter?  If this is a workaround, please add a comment.

File: creep.action.travelling.js [hunk 2]
```diff
@@ -8,9 +8,13 @@ action.step = function(creep){
     if(CHATTY) creep.say(this.name, SAY_PUBLIC);
     if( creep.target ){
         let pos;
-        if( creep.target.id == creep.id ) pos = new RoomPosition(25, 25, creep.data.travelRoom);
+        let targetRange = this.targetRange;
+        if( creep.target.id == creep.id ) {
+            targetRange = 24;
```
Comment (chobobobo) [inline creep.action.travelling.js]: This has the effect of making the creep regenerate its route once it gets inside the target room?

File: creep.behaviour.privateer.js (hunk 1 above)
Comment (karlthepagan) [inline creep.behaviour.privateer.js]: from what I have found this seems to be the standard mechanism to go to a room name

a target needs to be a real game object, and a room without vision is never a real game object

Comment (karlthepagan) [inline creep.behaviour.privateer.js]: @chobobobo see examples in `task.defense.js`

File: creep.action.travelling.js (hunk 2 above)
Comment (karlthepagan) [inline creep.action.travelling.js]: yup, also the only change i made was to let room to room travel have a 24 range

Comment (whardeman) [inline creep.action.travelling.js]: I was just wondering about setting targetRange slightly lower to ensure the creep moves away from the border before re-routing, to help prevent the NO PATH thrashing, but maybe that's been resolved elsewhere.

Comment (karlthepagan) [inline creep.action.travelling.js]: @whardeman resolved in the general case by #661

**Pull request merged**